esrally/driver/runner.py

```diff
@@ -1103,6 +1107,101 @@ def __repr__(self, *args, **kwargs):
+        templates = mandatory(params, "templates", self)
+        request_params = params.get("request-params", {})
+        for template, body in templates:
+            await es.cluster.put_component_template(name=template, body=body,
+                                                    params=request_params)
+        return len(templates), "ops"
+
+    def __repr__(self, *args, **kwargs):
+        return "create-component-template"
+
+
+class DeleteComponentTemplate(Runner):
+    """
+    Execute the `PUT index template API
+    <https://www.elastic.co/guide/en/elasticsearch/reference/current/indices-delete-component-template.html>`_.
+    """
+
+    async def __call__(self, es, params):
+        template_names = mandatory(params, "templates", self)
+        only_if_exists = params.get("only-if-exists", False)
+        request_params = params.get("request-params", {})
+
+        async def _exists(name):
+            return await es.transport.perform_request(
+                "HEAD", _make_path("_component_template", name)
+            )
+
+        ops_count = 0
+        for template_name in template_names:
+            if not only_if_exists:
+                await es.cluster.delete_component_template(name=template_name, params=request_params, ignore=[404])
+                ops_count += 1
+            elif only_if_exists and await _exists(template_name):
+                self.logger.info("Component Index template [%s] already exists. Deleting it.", template_name)
+                await es.cluster.delete_component_template(name=template_name, params=request_params)
+                ops_count += 1
+        return ops_count, "ops"
+
+    def __repr__(self, *args, **kwargs):
+        return "delete-component-template"
+
+
+class CreateComposableTemplate(Runner):
+    """
+    Execute the `PUT index template API <https://www.elastic.co/guide/en/elasticsearch/reference/current/indices-put-template.html>`_.
+    """
+
+    async def __call__(self, es, params):
+        templates = mandatory(params, "templates", self)
+        request_params = params.get("request-params", {})
+        for template, body in templates:
+            await es.cluster.put_index_template(name=template, body=body,
+                                                    params=request_params)
+        return len(templates), "ops"
+
+    def __repr__(self, *args, **kwargs):
+        return "create-composable-template"
+
+
+class DeleteComposableTemplate(Runner):
+    """
+    Execute the `PUT index template API <https://www.elastic.co/guide/en/elasticsearch/reference/current/indices-delete-template.html>`_.
+    """
+
+    async def __call__(self, es, params):
+        template_names = mandatory(params, "templates", self)
+        only_if_exists = params.get("only-if-exists", False)
+        request_params = params.get("request-params", {})
+        ops_count = 0
+
+        for template_name, delete_matching_indices, index_pattern in template_names:
```

Comment:
Given that this expands to a tuple I think `templates` would be a more appropriate name than `template_names`?